src/nomadic/util/metadata.py

```diff
@@ -105,14 +107,32 @@ class MetadataTableParser:
         ],
     }
 
-    def __init__(self, metadata_csv: str, include_unclassified: bool = True):
+    def __init__(self, metadata_path: str, include_unclassified: bool = True):
         """
         Load and sanity check the metadata table
 
         """
 
-        self.csv = metadata_csv
-        self.df = pd.read_csv(self.csv, delimiter=get_csv_delimiter(self.csv))
+        self.path = metadata_path
+        _, ext = os.path.splitext(self.path)
```

Comment:
I would suggest we encapsulate the code below in a private method, e.g. _load_metadata()